client/src/lib/dependency-injection/graphql/helpers.js

```diff
@@ -76,3 +86,112 @@ export const getFragments = ({ availableFragments, fragments = [] }) => (
         : capturedFragments
     ), '')
 );
+
+function getMatchingParamKey(matchWith, params) {
+  return Object.keys(params).find((paramName) => paramName.toLowerCase() === matchWith);
+}
+
+function getInputVariable({ templateName, singularName, params }) {
+  const inputKey = getMatchingParamKey('input', params);
+  if (inputKey) {
+    return `$${inputKey}: ${params[inputKey]}`;
+  }
+  // Graphql v3 fallbacks
+  switch (templateName) {
+    case CREATE:
+      return `$Input: ${getSingularName({ singularName })}CreateInputType!`;
+    case UPDATE:
+      return `$Input: ${getSingularName({ singularName })}UpdateInputType!`;
+    default:
+      throw new Error('Template is not a mutation - no input variable is required.');
+  }
+}
+
+function normaliseMutationParamsAndVars(paramsOrVars, dedup) {
+  return paramsOrVars.replace(dedup, '').replace(/(^\(+|\)+$)/mg, '');
+}
+
+export function getMutationVariables(obj) {
+  const inputVar = getInputVariable(obj);
+  switch (obj.templateName) {
+    case CREATE:
+      return `(${inputVar})`;
+    case UPDATE:
+      return `(
+        ${inputVar}
+        ${normaliseMutationParamsAndVars(getVariables(obj), inputVar)}
```

Comment:
If setting an input param like `input: 'CreateNoteInput!'` in the params object, we have to make sure to only output this _one_ for both variables and params.

Really I don't think update should allow for anything _other_ than the basic `input` (which would be consistent with what's happening for create) - but that functionality was there already so I can't just remove it.